src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -146,4 +152,14 @@ AttributeSyntax attribute
 		return attributeConstructorSymbol.ContainingType.ToDisplayString()
 			== "D2L.CodeStyle.Annotations.GenerateSyncAttribute";
 	}
+
+	public T UnhandledSyntax<T>( T node ) where T : SyntaxNode {
+		GeneratorError(
+			node.GetLocation(),
+			$"unhandled syntax kind: {node.Kind()}"
+		);
+
+		return node;

```

Comment:
We're forced by the nature of the recursion to return _something_, so we will just return the source text unmodified and stop recursing.

It will look a little weird in the generated output, but doing something else in all the cases would be a lot more work I think. It would possibly need to be different for every overload at least, e.g. `throw null` works for ExpressionSyntax, and if you wrap it in ExpressionStatement it could work for StatementSyntax, but etc. etc.

Reply:
We could also throw, but then that would stop adjacent recursions leading to a poor UX (you only get one diagnostic, which you fix and then get another diagnostic, repeat.... vs. just giving you all the diagnostics we reasonably can up-front.)